packages/ListBoxWithTags/src/ListBoxWithTags.js

```diff
@@ -6,98 +6,7 @@ import * as constants from "@paprika/constants/lib/Constants";
-import Tag, { Tags } from "@paprika/tag";
-import * as sc from "./ListBoxWithTags.styles";
-
-// TODO: Extract this to a separate component
-const renderTrigger = ({
-  allOptionsAreSelected,
-  hasError,
-  idListBoxContent,
-  isDisabled,
-  isReadOnly,
-  onRemove,
-  renderTag,
-  selectedOptions,
-  size,
-  tagLabelKey,
-  t,
-}) => (...args) => {
-  const [, , , attributes] = args;
-  const { propsForTrigger, refTrigger, dispatch, types, handleKeyDown, handleKeyUp, isOpen } = attributes;
-
-  function handleClick(event) {
-    event.stopPropagation();
-
-    if (event.key === "Enter" || event.key === " ") return;
-    dispatch({ type: types.togglePopover });
-  }
-
-  const handleRemove = option => () => {
-    onRemove(option);
-  };
-
-  function renderCaret(isDisabled) {
-    return isOpen ? <sc.UpIcon isDisabled={isDisabled} /> : <sc.DownIcon isDisabled={isDisabled} />;
-  }
-
-  const triggerProps = propsForTrigger();
-  const a11yTextOptions = selectedOptions.map(item => {
-    return tagLabelKey === null ? item.label : item[tagLabelKey];
-  });
-  const a11yText = selectedOptions.length === 0 ? t("listBoxWithTags.placeholder") : a11yTextOptions.join(", ");
-
-  return (
-    <>
-      <sc.Trigger
-        {...triggerProps}
-        aria-controls={idListBoxContent}
-        allOptionsAreSelected={allOptionsAreSelected}
-        hasError={hasError}
-        isDisabled={isDisabled || isReadOnly}
-        isReadOnly={isReadOnly}
-        onClick={handleClick}
-        onKeyUp={handleKeyUp}
-        onKeyDown={handleKeyDown}
-        ref={refTrigger}
-        size={size}
-      >
-        <Tags>
-          {selectedOptions.map(item => {
-            if (typeof renderTag === "function") {
-              return renderTag({ option: item, Tag, onRemove: handleRemove(item) });
-            }
-
-            const label = tagLabelKey === null ? item.label : item[tagLabelKey];
-
-            if (typeof label !== "string") {
-              throw Error(
-                `Your item ${JSON.stringify(
-                  item
-                )} must include the attribute "label", or you must indicate which attribute should be rendered as the label via the "tagLabelKey" prop.`
-              );
-            }
-
-            return (
-              <Tag as="li" key={label} onRemove={handleRemove(item)} size={size}>
-                {label}
-              </Tag>
-            );
-          })}
-          {selectedOptions.length ? null : (
-            <sc.Placeholder isDisabled={isDisabled} size={size}>
-              {t("listBoxWithTags.placeholder")}
-            </sc.Placeholder>
-          )}
-        </Tags>
-        {allOptionsAreSelected ? null : renderCaret(isDisabled)}
-      </sc.Trigger>
-      <sc.TriggerLabel id={`${triggerProps.id}__label`}>
-        {t("listBoxWithTags.a11y_text_trigger", { options: a11yText })}
-      </sc.TriggerLabel>
-    </>
-  );
-};
+import renderTrigger from "./renderTrigger";
```

Comment:
At this point `renderTrigger` can pretty much be a component:

```js
function ListBoxTrigger({ allOptionsAreSelected, ....... } )
```

```js
import ListBoxTrigger from "./ListBoxTrigger";
```

Reply:
IMO, we need to reserve `renderSomthing` functions for the render props OR internal render functions. If the function is already extracted, then it is technically a component already.

Reply:
It is a render prop though, specifically for the `children` prop of `ListBox.Trigger`.  This needs to return a function, not JSX, or it won't be used at all:

https://github.com/acl-services/paprika/blob/master/packages/ListBox/src/components/Trigger/Trigger.js#L245

My original idea was to extract as a component, but a component can't return a function (as far as I know!).  

I still think it has a specific enough purpose that it should be extracted from `ListBoxWithTags.js` though 😅